File: practica2/c++/mul.cc
#include <stack>
#include "mul.h"
#include <string>

using namespace std;

void mul::ejecutar(stack<int32_t>& pila, int32_t& pc) const
{
	int32_t op1 = pila.top();
	pila.pop();
	int32_t op2=pila.top();
	pila.pop();
	pila.push(op1*op2);
	pc++;
}

string mul::nombre() const
{
	return "mul";
}

File: practica2/java/mul.java

/**
 * mul
 */
class mul extends inst{
    public void ejecutar(Stack<int32_t> pila, int32_t pc){
        int32_t op1 = pila.peek();
	    pila.pop();
	    int32_t op2=pila.peek();
	    pila.pop();
	    pila.push(op1*op2);
	    pc++;
    }
    public String nombre(){
        return "mul";
    }

}